packages/usa-date-picker/src/index.js

```diff
@@ -1785,24 +1816,43 @@ const handleShiftPageDownFromDate = adjustCalendar((date) => addYears(date, 1));
 const handleShiftPageUpFromDate = adjustCalendar((date) => subYears(date, 1));
 
 /**
- * display the calendar for the mouseover date.
+ * Set range date classes without re-rendering the calendar. Called when date button is hovered.
+ * Returns early if the date hovered is disabled or if there is a selected date already.
  *
- * @param {MouseEvent} event The mouseover event
- * @param {HTMLButtonElement} dateEl A date element within the date picker component
+ * @param {HTMLElement} dateEl - Calendar date button within the date picker component.
  */
+
 const handleMouseoverFromDate = (dateEl) => {
   if (dateEl.disabled) return;
 
-  const calendarEl = dateEl.closest(DATE_PICKER_CALENDAR);
+  const hoverDate = parseDateString(dateEl.dataset.value);
+  const { calendarEl, selectedDate, rangeDate } = getDatePickerContext(dateEl);
 
-  const currentCalendarDate = calendarEl.dataset.value;
-  const hoverDate = dateEl.dataset.value;
+  if (selectedDate) return;
 
-  if (hoverDate === currentCalendarDate) return;
+  const { withinRangeStartDate, withinRangeEndDate } = setRangeDates(
+    hoverDate,
+    rangeDate,
+  );
 
-  const dateToDisplay = parseDateString(hoverDate);
-  const newCalendar = renderCalendar(calendarEl, dateToDisplay);
-  newCalendar.querySelector(CALENDAR_DATE_FOCUSED).focus();
+  const dateButtons = calendarEl.querySelectorAll(
+    `.${CALENDAR_DATE_CURRENT_MONTH_CLASS}`,
+  );
+
+  dateButtons.forEach((button) => {
+    const buttonDate = parseDateString(button.dataset.value);
+    if (
+      isDateWithinMinAndMax(
+        buttonDate,
+        withinRangeStartDate,
+        withinRangeEndDate,
+      )
+    ) {
+      button.classList.add(CALENDAR_DATE_WITHIN_RANGE_CLASS);
+    } else {
+      button.classList.remove(CALENDAR_DATE_WITHIN_RANGE_CLASS);
+    }
```

Comment:
For each date button in the current calendar month, check if it's value is within the date range.

If it is, add the `within range` class. If itis not, remove it.